alica_tracing/include/tracing/AlicaTrace.h

```diff
@@ -0,0 +1,197 @@
+#pragma once
+
+#include <exception>
+#include <memory>
+#include <string>
+
+#include <engine/IAlicaTrace.h>
+#include <jaegertracing/Tracer.h>
+#include <ros/ros.h>
+
+namespace alicaTracing
+{
+
+using RawTrace = opentracing::Span;
+using RawTracePtr = std::unique_ptr<RawTrace>;
+// Value can be numeric types, strings, or bools.
+using RawTraceValue = opentracing::Value;
+using TraceContext = std::string;
+
+class Trace : public alica::IAlicaTrace
+{
+public:
+    // Create partial trace instance with optional reference to the parent.
+    // Trace without parent will be reported as root of the trace and can have further children.
+    // Although library doesn't validate operation name or stop any operation to be a root,
+    // users of the library are supposed to follow this specs
+    // https://app.diagrams.net/?libs=general;basic;arrows#G17zQOPcT-Tr2GStr2nYzvcbgGe_SeFUEi
+    // Creation of this instance will set operation start time to now()
+    Trace(const std::string& opName, std::optional<const std::string> parent = std::nullopt)
+    {
+        if (parent) {
+            std::stringstream ss(parent.value());
+            jaegertracing::SpanContext parentCtx = jaegertracing::SpanContext::fromStream(ss);
+            _rawTrace = opentracing::Tracer::Global()->StartSpan(opName, {opentracing::ChildOf(&parentCtx)});
+        } else {
+            _rawTrace = opentracing::Tracer::Global()->StartSpan(opName);
+            if (opName == "MasterPlan") {
```

Comment:
put a comment here that this is required so that the master trace is reported immediately